### PR TITLE
Move window controls so not occluded in full screen

### LINKED ISSUE #74: Titlebar not correctly shown in full screen mode
When maximized the title bar is partially hidden under the menubar. This makes unmaximize pretty hard.

bildschirmfoto 2015-04-14 um 16 12 22

### PR DESCRIPTION
**Pull request by mattbroussard**
Now, window control buttons are usable in full screen (issue #74). In non-full screen, they're also moved to the side so they don't block the gear icon.

screen shot 2015-04-14 at 11 59 31 pm

screen shot 2015-04-14 at 11 59 41 pm

### CODE REVIEW

File: Messenger/AppDelegate.mm [hunk 1]
```diff
@@ -163,7 +164,9 @@ - (void)updateWindowTitlebar {
   auto updateButton = ^(NSView* buttonView) {
     auto buttonFrame = buttonView.frame;
     buttonFrame.origin.y = round((kTitlebarHeight - buttonFrame.size.height) / 2.0);
+    if (fullScreen) buttonFrame.origin.y -= 15;
```
Comment (rsms) [inline Messenger/AppDelegate.mm]: Can you please change this to not use a constant but instead `kTitlebarHeight`?

File: Messenger/AppDelegate.mm [hunk 2]
```diff
@@ -163,7 +164,9 @@ - (void)updateWindowTitlebar {
...
     buttonFrame.origin.x = x;
+    if (!fullScreen) buttonFrame.origin.x += 35;
```
Comment (rsms) [inline Messenger/AppDelegate.mm]: Same here, use `kTitlebarHeight` instead of constant

**Comment by mattbroussard** [comment]
Added named constants for the values, and comments explaining the math. Since neither value is really related to kTitlebarHeight (the X offset in non-fullscreen is related to the position/size of the gear icon, and the Y offset in fullscreen is related to the fixed-size fullscreen titlebar, whose size is not governed by kTitlebarHeight), it wasn't appropriate to use that.

Does this look good or do you think there's a better way we can structure this?

**Comment by mattbroussard** [comment]
Oh, I just noticed in master now there's code that hides the gear icon and puts those options instead in the app menu. Will update this shortly to remove the x-offset.
screen shot 2015-04-16 at 12 25 45 am

**Comment by rsms** [comment]
Nice work

**Pull request merged**